classify-cluster-priority/README.md

```diff
@@ -0,0 +1,85 @@
+# Classifying clusters based on priority
+
+Contents:
+
+1. User stories
+2. Current practice
+3. Proposal
+4. Considerations and explanations
+5. Possible applications
+
+## User stories
+
+- As a cluster admin, I want to easily communicate the purpose and importance of a cluster to those interacting with it, so that they can act accordingly.
+- As Giant Swarm operations staff I want to understand the importance of clusters so I can make appropriate decisions, e. g. when prioritizing problem mitigation in an incident affecting multiple clusters.
+
+## Current practice
+
+- One customer uses the first character of the cluster name to distinguish between production (`p`), development (`d`), and staging (`s`). This customer also has clusters which don't fall into either of these categories and their names start with arbitrary letters.
+- Many customers use specific terms and phrases in the cluster description, e. g. "Production", "test", "testing", "ignore alerts", "prd", "prod", "nonprod".
+- Some customers use labels on the `clusters.cluster.x-k8s.io` resource (the cluster's main resource) indicating the importance, e.g:
+    - `odp/environment: development`
+    - `odp/environment: staging`
+    - `odp/environment: production`.
+
+## Proposal
+
+We introduce a label and a schema for the label values to indicate the relative importance of a cluster.
+
+Label name: `giantswarm.io/service-priority`
+
+The label is to be used on the main resource defining the cluster, usually of resource type `clusters.cluster.x-k8s.io`.
+
+To simplify maintenance and avoid logical conflicts, this label SHOULD NOT be applied on other resources of the cluster. Systems MUST ignore the label used on other resources in the same cluster. This means that it is not possible, for example, to differentiate between the importance of several node pools within the same cluster.
+
+### Well-defined label values
```

Comment:
This whole concept slightly me of how log levels are usually implemented. Instead of fix values we could define a scale of let's say 0-1000 for simplicity (usually this is much wider I think). Then we can name certain point for example: { 250: Standard, 500: Important, 750: Critical }. This still gives room to make a difference between critical and critical and even makes the system easily extendable in the future if the spaces are big enough.

Labels could still be by both numeric value and human readable value, CLI tools could rely on the standard named levels and assign the proper labels and values.

The label values and the they are in sync can be enforced via admission and validation webhooks I think.

Reply:
If you are suggesting a numeric value system, I am not a fan of it for the reason that numbers, especially when it comes to ranking priority, are not self-explanatory. As a user, many will wonder: is 1 a high priority or a low one? (Many people in day to day communication say things like "this is priority one for me". But it's also a low number compared to 2 and 3. So people will always have to look up some reference.

I think natural strings like we have here are easier to get, while of course there is not guarantee that they work for every user.

Reply:
Yeah I’m not keen on under scale either. Customers won’t know what value to set and therefore just leave it as default.